File: BEST-App-main/lib/model/product.dart
class Product {
  final int id;
  final int brandId;
  final int categoryId;
  final int shopId;
  final String name;
  final int quantity;
  final double price;
  final String description;
  final String code;
  final int discount;
  final int sold;
  final String createdAt;
  final int status;

  Product(
      this.id,
      this.brandId,
      this.categoryId,
      this.shopId,
      this.name,
      this.quantity,
      this.price,
      this.description,
      this.code,
      this.discount,
      this.sold,
      this.createdAt,
      this.status);

  static Map<String, dynamic> toJson(Product product) {
    return {
      'id': product.id,
      'brand_id': product.brandId,
      'category_id': product.categoryId,
      'shop_id': product.shopId,
      'name': product.name,
      'quantity': product.quantity,
      'price': product.price,
      'description': product.description,
      'code': product.code,
      'discount': product.discount,
      'sold': product.sold,
      'created_at': product.createdAt,
      'status': product.status
    };
  }
}

File: spring-best/src/main/java/com/best/springbest/model/Product.java
package com.best.springbest.model;

import com.sun.istack.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.format.annotation.DateTimeFormat;

import javax.persistence.*;
import java.util.Date;

@Entity
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "products")
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name="brand_id",nullable = false)
    private Brand brand;

    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name="category_id",nullable = false)
    private Category category;

    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name="shop_id",nullable = false)
    private Shop shop;

    @NotNull
    private String name;

    private int quantity;

    private double price;

    private String description;

    private String code;

    private int discount;

    private int sold;

    @DateTimeFormat(pattern = "yyyy-MM-dd")
    private Date created_at;

    private int status;
}
